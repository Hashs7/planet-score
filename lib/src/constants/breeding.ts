/**
 *
 * @type {{MEDIUM: string, BAD: string, GOOD: string}}
 */
export enum BREEDING_CONDITION {
  GOOD = 'v',
  MEDIUM = 'o',
  BAD = 'r'
}

/**
 * Color state for breeding score
 *
 * @type {{[p: string]: string}}
 */
export const BREEDING_CONDITION_COLOR: { [index: string]: string } = {
  [BREEDING_CONDITION.GOOD]: '#007C3B',
  [BREEDING_CONDITION.MEDIUM]: '#F07E02',
  [BREEDING_CONDITION.BAD]: '#E33116'
};

/**
 * List of available animals
 *
 * @type {{OTHER: string, BEE: string, FARMING_FISH: string, FISHING_FISH: string, CHICKEN: string, SHEEP: string, DUCK: string, COW: string, GOAT: string, RABBIT: string, PIG: string}}
 */
export enum BREEDING_ANIMAL {
  COW = 'bovin',
  CHICKEN = 'poule',
  SHEEP = 'brebis',
  GOAT = 'chevre',
  PIG = 'porc'

  // Icon not integrated
  // BEE = 'abeille',
  // DUCK = 'canard',
  // RABBIT = 'lapin',
  // FARMING_FISH = 'poisson_aquaculture',
  // FISHING_FISH = 'poisson_peche',
  // OTHER = 'autre'
}

/**
 * Regex to extract only breeding animal word
 *
 * @type {RegExp}
 */
export const BREEDING_ANIMAL_REGEX = new RegExp(`(${Object.values(BREEDING_ANIMAL).join('|')})`);
