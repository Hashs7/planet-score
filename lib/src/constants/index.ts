export * from './breeding';
export * from './score';
