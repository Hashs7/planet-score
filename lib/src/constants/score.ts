/**
 * The score values.
 */
export enum ScoreValues {
  A = 'A',
  B = 'B',
  C = 'C',
  D = 'D',
  E = 'E'
}

/**
 * The score values in order of best to worst.
 */
export const SCORE_POSITIONS = [
  ScoreValues.A,
  ScoreValues.B,
  ScoreValues.C,
  ScoreValues.D,
  ScoreValues.E
];
