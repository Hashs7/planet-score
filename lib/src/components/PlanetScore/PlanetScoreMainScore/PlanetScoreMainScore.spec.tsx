import { render } from '@testing-library/react';
import PlanetScoreMainScore from './PlanetScoreMainScore';
import { SCORE_POSITIONS } from '@/constants';
import { SvgWrapper } from '@tests/utils';

describe('<PlanetScoreMainScore>', () => {
  it(`should not renders with invalid score`, () => {
    const { container } = render(<PlanetScoreMainScore score="" />);

    expect(container).toMatchSnapshot();
  });

  SCORE_POSITIONS.forEach((score) => {
    it(`(${score}) renders ${score} circle with associated color`, () => {
      const { container } = render(<PlanetScoreMainScore score={score} />, { wrapper: SvgWrapper });

      expect(container).toMatchSnapshot();
    });
  });
});
