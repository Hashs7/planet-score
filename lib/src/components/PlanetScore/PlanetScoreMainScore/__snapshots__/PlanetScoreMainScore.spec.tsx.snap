// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetScoreMainScore> (A) renders A circle with associated color 1`] = `
<div>
  <svg>
    <g
      id="score-A"
      transform="translate(-430 0)"
    >
      <ellipse
        cx="1001.8"
        cy="219.4"
        rx="74.6"
        ry="74.6"
        style="fill: #027D3D;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 138.2766 772.6507)"
      />
      <path
        d="M1058.8,162.4c-15.2-15.2-35.5-23.6-57-23.6c-21.5,0-41.8,8.4-57,23.6
            c-15.2,15.2-23.6,35.5-23.6,57s8.4,41.8,23.6,57c15.2,15.2,35.5,23.6,57,23.6c21.5,0,41.8-8.4,57-23.6
            c15.2-15.2,23.6-35.5,23.6-57S1074,177.6,1058.8,162.4z M927.2,219.4c0-41.2,33.4-74.6,74.6-74.6c41.2,0,74.6,33.4,74.6,74.6
            s-33.4,74.6-74.6,74.6C960.6,294.1,927.2,260.6,927.2,219.4z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M959.7,253.7l27.7-74.8c1.3-3.4,3.2-6.2,5.8-8.3c2.6-2.1,5.5-3.1,8.9-3.1c3.5,0,6.6,1,9.1,3.1
            c2.5,2.1,4.4,4.8,5.6,8.3l27,74.8c0.7,2,1.1,3.7,1.1,5.1c0,1.7-0.4,3.3-1.3,4.7c-0.9,1.4-2,2.5-3.5,3.2c-1.4,0.8-3,1.1-4.7,1.1
            c-5.1,0-8.5-2.4-10.1-7.1l-4.1-12.9H983l-4.2,12.8c-1.5,4.8-4.8,7.2-9.9,7.2c-2.8,0-5.2-0.9-7.2-2.8c-2-1.9-3-4.1-3-6.8
            C958.8,257,959.1,255.4,959.7,253.7z M1001.9,189l-13,39.9h26.1l-12.8-39.9H1001.9z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreMainScore> (B) renders B circle with associated color 1`] = `
<div>
  <svg>
    <g
      id="score-B"
      transform="translate(-320 0)"
    >
      <g>
        <ellipse
          cx="1001.8"
          cy="219.4"
          rx="74.6"
          ry="74.6"
          style="fill: #7AB72A;"
          transform="matrix(0.7071 -0.7071 0.7071 0.7071 138.2766 772.6507)"
        />
        <path
          d="M1058.8,162.4c-15.2-15.2-35.5-23.6-57-23.6c-21.5,0-41.8,8.4-57,23.6
			c-15.2,15.2-23.6,35.5-23.6,57s8.4,41.8,23.6,57c15.2,15.2,35.5,23.6,57,23.6c21.5,0,41.8-8.4,57-23.6
			c15.2-15.2,23.6-35.5,23.6-57S1074,177.6,1058.8,162.4z M927.2,219.4c0-41.2,33.4-74.6,74.6-74.6c41.2,0,74.6,33.4,74.6,74.6
			s-33.4,74.6-74.6,74.6C960.6,294.1,927.2,260.6,927.2,219.4z"
          style="fill: #FFFFFF;"
        />
      </g>
      <g>
        <path
          d="M968.7,256.6V180c0-3.2,1-5.7,3-7.5c2-1.8,4.6-2.7,7.8-2.7h26.2c4.7,0,9,1,12.9,3
			c3.9,2,7,4.8,9.3,8.5c2.3,3.7,3.4,7.8,3.4,12.5c0,9.5-4.2,16.2-12.7,20.1v0.3c2.8,0.4,5.5,1.3,8.1,2.6c2.5,1.4,4.8,3.1,6.7,5.2
			c1.9,2.1,3.4,4.4,4.4,7c1,2.6,1.5,5.4,1.5,8.5c0,9.4-2.9,16.7-8.6,21.9c-5.8,5.2-13.4,7.7-23,7.7h-28.1c-3.2,0-5.8-1-7.9-2.9
			C969.7,262.3,968.7,259.8,968.7,256.6z M989.2,187.8v20.5h10.7c3.4,0,6.1-0.9,8.1-2.8c2-1.9,3-4.5,3-7.9c0-3.1-1-5.5-3.1-7.2
			c-2.1-1.7-4.7-2.5-8-2.5H989.2z M989.2,226.3V249h16.1c3.8,0,6.9-1.1,9.3-3.2c2.4-2.1,3.6-5.1,3.6-8.8c0-3.3-1.2-5.9-3.7-7.9
			c-2.5-1.9-6.1-2.9-10.8-2.9H989.2z"
          style="fill: #FFFFFF;"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreMainScore> (C) renders C circle with associated color 1`] = `
<div>
  <svg>
    <g
      id="score-C"
      transform="translate(-210 0)"
    >
      <ellipse
        cx="1001.8"
        cy="219.4"
        rx="74.6"
        ry="74.6"
        style="fill: #FBC404;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 138.2766 772.6507)"
      />
      <path
        d="M1058.8,162.4c-15.2-15.2-35.5-23.6-57-23.6c-21.5,0-41.8,8.4-57,23.6
          c-15.2,15.2-23.6,35.5-23.6,57s8.4,41.8,23.6,57c15.2,15.2,35.5,23.6,57,23.6c21.5,0,41.8-8.4,57-23.6
          c15.2-15.2,23.6-35.5,23.6-57S1074,177.6,1058.8,162.4z M927.2,219.4c0-41.2,33.4-74.6,74.6-74.6c41.2,0,74.6,33.4,74.6,74.6
          s-33.4,74.6-74.6,74.6C960.6,294.1,927.2,260.6,927.2,219.4z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M958.4,218.1c0-7.1,1.1-13.8,3.4-19.9c2.3-6.2,5.5-11.6,9.8-16.2c4.2-4.6,9.2-8.2,15-10.7
          c5.8-2.5,12.1-3.8,19.1-3.8c2.8,0,6,0.3,9.5,0.8c3.5,0.5,6.7,1.4,9.7,2.5c3,1.1,5.5,2.7,7.5,4.7c2,2,3,4.5,3,7.4
          c0,2.5-0.8,4.7-2.5,6.5c-1.6,1.8-3.7,2.7-6.1,2.7c-1.5,0-3.5-0.5-6.2-1.5c-2.7-1-5-1.8-6.9-2.4c-1.9-0.6-4.6-0.9-8-0.9
          c-4.2,0-7.9,0.8-11.1,2.5c-3.2,1.6-5.9,3.9-8.1,6.7c-2.1,2.8-3.8,6.1-4.9,9.9c-1.1,3.8-1.7,7.9-1.7,12.2c0,8.6,2.3,15.9,6.8,21.7
          c4.6,5.8,10.9,8.7,19,8.7c3.4,0,6.1-0.3,8.1-1c2-0.7,4.4-1.6,7.3-2.9c2.8-1.3,5-1.9,6.5-1.9c2.8,0,5,1.1,6.6,3.2
          c1.6,2.1,2.4,4.2,2.4,6.1c0,4-1.7,7.3-5.2,9.7c-3.4,2.4-7.6,4.1-12.3,5.1c-4.8,1-9.3,1.5-13.5,1.5c-9.1,0-17.3-2.2-24.5-6.7
          s-12.8-10.6-16.8-18.3C960.4,236,958.4,227.5,958.4,218.1z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreMainScore> (D) renders D circle with associated color 1`] = `
<div>
  <svg>
    <g
      id="score-D"
      transform="translate(-105 0)"
    >
      <ellipse
        cx="1001.8"
        cy="219.4"
        rx="74.6"
        ry="74.6"
        style="fill: #ED7A06;"
        transform="matrix(0.7071 -0.7071 0.7071 0.7071 138.2766 772.6507)"
      />
      <path
        d="M1058.8,162.4c-15.2-15.2-35.5-23.6-57-23.6c-21.5,0-41.8,8.4-57,23.6
          c-15.2,15.2-23.6,35.5-23.6,57s8.4,41.8,23.6,57c15.2,15.2,35.5,23.6,57,23.6c21.5,0,41.8-8.4,57-23.6
          c15.2-15.2,23.6-35.5,23.6-57S1074,177.6,1058.8,162.4z M927.2,219.4c0-41.2,33.4-74.6,74.6-74.6c41.2,0,74.6,33.4,74.6,74.6
          s-33.4,74.6-74.6,74.6C960.6,294.1,927.2,260.6,927.2,219.4z"
        style="fill: #FFFFFF;"
      />
      <path
        d="M964,256.7v-77c0-3.1,1-5.7,3-7.7c2-2,4.5-3,7.6-3h22.8c7.7,0,14.6,1.2,20.6,3.6
          c6,2.4,11.1,5.7,15.2,10.1c4.1,4.4,7.3,9.6,9.4,15.8c2.1,6.2,3.2,13.1,3.2,20.8c0,7.2-1.1,13.7-3.3,19.5
          c-2.2,5.9-5.4,10.9-9.6,15.1c-4.2,4.2-9.2,7.4-15,9.6c-5.9,2.2-12.2,3.4-19.1,3.4h-23.9c-1.7,0-3.4-0.3-5-1
          c-1.6-0.7-3-1.8-4.2-3.3C964.6,261.2,964,259.2,964,256.7z M984.7,188.2V248h13.9c5.8,0,10.6-1.3,14.5-3.9
          c3.9-2.6,6.7-6.1,8.6-10.5c1.9-4.4,2.8-9.5,2.8-15.2c0-5.9-1-11.2-2.9-15.8c-1.9-4.6-4.9-8.2-9-10.8c-4.1-2.6-9.2-3.9-15.3-3.9
          H984.7z"
        style="fill: #FFFFFF;"
      />
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreMainScore> (E) renders E circle with associated color 1`] = `
<div>
  <svg>
    <g
      id="score-E"
    >
      <g>
        <ellipse
          cx="1001.8"
          cy="219.4"
          rx="74.6"
          ry="74.6"
          style="fill: #E33116;"
          transform="matrix(0.7071 -0.7071 0.7071 0.7071 138.2766 772.6507)"
        />
        <path
          d="M1058.8,162.4c-15.2-15.2-35.5-23.6-57-23.6c-21.5,0-41.8,8.4-57,23.6
			c-15.2,15.2-23.6,35.5-23.6,57s8.4,41.8,23.6,57c15.2,15.2,35.5,23.6,57,23.6c21.5,0,41.8-8.4,57-23.6
			c15.2-15.2,23.6-35.5,23.6-57S1074,177.6,1058.8,162.4z M927.2,219.4c0-41.2,33.4-74.6,74.6-74.6c41.2,0,74.6,33.4,74.6,74.6
			s-33.4,74.6-74.6,74.6C960.6,294.1,927.2,260.6,927.2,219.4z"
          style="fill: #FFFFFF;"
        />
      </g>
      <g>
        <path
          d="M975.8,256.7v-75.6c0-3.6,0.9-6.5,2.8-8.7c1.8-2.2,4.5-3.3,8-3.3h34.1c3.1,0,5.5,0.9,7.2,2.8
			c1.7,1.9,2.5,4.1,2.5,6.7c0,1.7-0.4,3.3-1.1,4.7c-0.8,1.4-1.9,2.6-3.3,3.5c-1.5,0.9-3.2,1.3-5.3,1.3h-24.2v20h22.6
			c3.2,0,5.7,0.9,7.5,2.8c1.7,1.8,2.6,4.1,2.6,6.8c0,2.5-0.9,4.7-2.7,6.6c-1.8,1.9-4.3,2.9-7.4,2.9h-22.6V248h25.4
			c3.2,0,5.6,0.9,7.3,2.8c1.7,1.9,2.5,4.1,2.5,6.7c0,1.5-0.4,3-1.1,4.5c-0.8,1.5-1.9,2.7-3.4,3.6c-1.5,0.9-3.2,1.3-5.3,1.3h-35.5
			c-3,0-5.6-1-7.6-2.9C976.8,262.3,975.8,259.8,975.8,256.7z"
          style="fill: #FFFFFF;"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`<PlanetScoreMainScore> should not renders with invalid score 1`] = `<div />`;
